pandas/util/testing.py

```diff
@@ -536,23 +537,128 @@ def assert_equal(a, b, msg=""):
     assert a == b, "%s: %r != %r" % (msg.format(a,b), a, b)
 
 
-def assert_index_equal(left, right, exact=False, check_names=True):
+def assert_index_equal(left, right, exact=False, check_names=True,
+                       check_less_precise=False, check_exact=True, obj='Index'):
+    """Check that left and right Index are equal.
+
+    Parameters
+    ----------
+    left : Index
+    right : Index
+    exact : bool, default False
+        Whether to check the Index class, dtype and inferred_type are identical.
+    check_names : bool, default True
+        Whether to check the names attribute.
+    check_less_precise : bool, default False
+        Specify comparison precision. Only used when check_exact is False.
+        5 digits (False) or 3 digits (True) after decimal points are compared.
+    check_exact : bool, default True
+        Whether to compare number exactly.
```

Comment:
After I put this comment, I saw that this is the way it is done for assert_frame/series_equal as well, so at least it is consistent. 
But apart from that, I find it not really a nice interface. But as the same is true for assert_frame/series_equal, this can be deferred to a separate issue.

In short, this way of specifying the precision is not very flexible. I think we should try to rely on something like `np.testing.assert_allclose` with relative and absolute tolerance.